### ISSUE TITLE
KDF-108 SPEC: Change algSpecs to capabilities

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#386]
The server returns an error during registering of the following request:

```
        {
          "algorithm":"KDF",
          "prereqVals":[
            {
              "algorithm":"HMAC",
              "valValue":"same"
            },
            {
              "algorithm":"SHA",
              "valValue":"same"
            }
          ],
          "algSpecs":[
            {
              "kdfMode":"double pipeline iteration",
              "macMode":[
                "HMAC-SHA-1",
                "HMAC-SHA2-224",
                "HMAC-SHA2-256",
                "HMAC-SHA2-384",
                "HMAC-SHA2-512",
                "CMAC-AES128",
                "CMAC-AES192",
                "CMAC-AES256"
              ],
              "supportedLengths":[
                1,
                77,
                128,
                789,
                3456,
                4096
              ],
              "fixedDataOrder":[
                "after fixed data"
              ],
              "counterLength":[
                32
              ],
              "supportsEmptyIv":true
            }
          ]
        }
```

This also applies to counter and feedback mode.

The error is:
```
{
  "acv_version" : "0.4",
  "error" : "Incorrectly formatted JSON (1:1905): expected field name was not provided: capabilities"
}
```

**Comment by celic** [comment]
This should work if you rename `algSpecs` to `capabilities` and is a mismatch in our server code.

**Comment by smuellerDD** [comment]
Am Montag, 4. Juni 2018, 18:04:49 CEST schrieb Chris Celi:

Hi Chris,

> This should work if you rename `algSpecs` to `capabilities` and is a
> mismatch in our server code.

Yes, it works. But now I get an error with a wrong string:

```
{
  "acv_version" : "0.4",
  "error" : "Property value(HMAC-SHA-1) for (macMode) in Algorithm (KDF 
(null)) is not valid. Validation: (Values must be one of: CMAC-AES128, CMAC-
AES192, CMAC-AES256, CMAC-TDES, HMAC-SHA1, HMAC-SHA224, HMAC-SHA256, HMAC-
SHA384, HMAC-SHA512)"
}
```

I guess the server code should be updated to use HMAC-SHA-1 to be consistent 
with other algos.

Ciao
Stephan

**Comment by celic** [comment]
The server has been updated to address this locally. I'll update this ticket when it is deployed.

**Comment by smuellerDD** [comment]
Fixed with pull request #417.
